src/libpriv/rpmostree-core.c

```diff
@@ -1101,11 +1101,18 @@ rpmostree_context_download_metadata (RpmOstreeContext *self,
       return TRUE;
     }
 
-  g_autoptr(GString) enabled_repos = g_string_new ("Enabled rpm-md repositories:");
-  for (guint i = 0; i < rpmmd_repos->len; i++)
+  g_autoptr(GString) enabled_repos = g_string_new ("");
+  if (rpmmd_repos->len > 0)
     {
-      DnfRepo *repo = rpmmd_repos->pdata[i];
-      g_string_append_printf (enabled_repos, " %s", dnf_repo_get_id (repo));
+      for (guint i = 0; i < rpmmd_repos->len; i++)
```

Comment:
Aren't we missing e.g.

```suggestion
      g_string_append (enabled_repos, "Enabled rpm-md repositories:");
      for (guint i = 0; i < rpmmd_repos->len; i++)
```

?

A slight tweak would be to initialize `enabled_repos` to `NULL` and do `g_string_new(<static string literal>)` instead of `g_string_append` in both branches.

Reply:
Thanks, yep!